nixos/modules/services/databases/redis.nix

```diff
@@ -208,21 +222,26 @@ in
 
     environment.systemPackages = [ cfg.package ];
 
-    systemd.services.redis =
-      { description = "Redis Server";
+    systemd.services.redis = {
+      description = "Redis Server";
 
-        wantedBy = [ "multi-user.target" ];
-        after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network.target" ];
 
-        serviceConfig = {
-          ExecStart = "${cfg.package}/bin/redis-server ${redisConfig}";
-          RuntimeDirectory = "redis";
-          StateDirectory = "redis";
-          Type = "notify";
-          User = "redis";
-        };
-      };
+      preStart = ''
+        install -m 600 ${redisConfig} /run/redis/redis.conf
+      '' + optionalString (cfg.requirePassFile != null) ''
+        password=$(cat ${escapeShellArg cfg.requirePassFile})
```

Comment:
Is `escapeShellArg` used instead of quoting the filename? I think this would be more readable: `$(cat "${cfg.requirePassFile}")`

Reply:
Yes. `$(cat "${cfg.requirePassFile}")` could in theory be a problem if requirePassFile contains a `"`. Should ignore this case?

Reply:
Didn't think of that. Never mind then!